### PR TITLE
[RISCV] Support Xsfvfwmaccqqq extensions

### PR DESCRIPTION
**Pull request by 4vtomat**
Bfloat16 Matrix Multiply Accumulate Instruction
https://sifive.cdn.prismic.io/sifive/c391d53e-ffcf-4091-82f6-c37bf3e883ed_xsfvfwmaccqqq-spec.pdf

### CODE REVIEW

**Comment by 4vtomat** [comment]
ping

File: llvm/lib/Target/RISCV/RISCVFeatures.td [hunk 1]
```diff
@@ -813,6 +813,14 @@ def HasVendorXSfcie : Predicate<"Subtarget->hasVendorXSfcie()">,
                         AssemblerPredicate<(all_of FeatureVendorXSfcie),
                         "'XSfcie' (SiFive Custom Instruction Extension SCIE.)">;
 
+def FeatureVendorXSfvfwmaccqqq
+    : SubtargetFeature<"xsfvfwmaccqqq", "HasVendorXSfvfwmaccqqq", "true",
+                       "'XSfvfwmaccqqq' (SiFive Matrix Multiply Accumulate Instruction and 4-by-4))",
+                       [FeatureStdExtZve32x]>;
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVFeatures.td:819]: Should this be Zve32f since the return type is FP?

File: llvm/lib/Support/RISCVISAInfo.cpp [hunk 2]
```diff
@@ -991,6 +992,7 @@ static const char *ImpliedExtsF[] = {"zicsr"};
 static const char *ImpliedExtsV[] = {"zvl128b", "zve64d"};
 static const char *ImpliedExtsXTHeadVdot[] = {"v"};
 static const char *ImpliedExtsXsfvcp[] = {"zve32x"};
+static const char *ImpliedExtsXsfvfwmaccqqq[] = {"zve32x"};
```
Comment (topperc) [inline llvm/lib/Support/RISCVISAInfo.cpp:995]: Zve32f?

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**